src/rwe/GameScene.cpp

```diff
@@ -859,29 +903,32 @@ namespace rwe
                 // (clamped to map bounds)
 
                 auto minimapToWorld = minimapToWorldMatrix(simulation.terrain, minimapRect);
-                auto mousePos = getMousePosition();
-                auto worldPos = minimapToWorld * Vector3f(static_cast<float>(mousePos.x) + 0.5f, static_cast<float>(mousePos.y) + 0.5, 0.0f);
+                auto mousePos       = getMousePosition();
+                auto worldPos
+                    = minimapToWorld
+                      * Vector3f(static_cast<float>(mousePos.x) + 0.5f, static_cast<float>(mousePos.y) + 0.5, 0.0f);
                 auto newCameraPos = cameraConstraint.clamp(Vector2f(worldPos.x, worldPos.z));
                 camera.setPosition(Vector3f(newCameraPos.x, camera.getRawPosition().y, newCameraPos.y));
             }
         }
 
-        hoveredUnit = getUnitUnderCursor();
+        elementOfInterest = getElementOfInterest();
```

Comment:
Terrain position and units are both and the only elements the player can interacts with directly or indirectly, did I get it right?